### PR TITLE
Protocol plumbing simplifications

### PR DESCRIPTION
**Pull request by dcoutts**
A large number of small incremental changes to improve the plumbing of the protocol information in the node top level and CLI.

With simpler plumbing it will be a lot easier to plumb the Shelley things in.

### CODE REVIEW

**Comment by dcoutts** [comment]
Note that this will not yet build because it depends on https://github.com/input-output-hk/ouroboros-network/pull/1927 which simplifies some of the plumbing on the consensus side.

This will need to be rebased on #774 once that is merged. The first two patches are from that.

File: cardano-config/src/Cardano/Config/Protocol.hs [hunk 1]
```diff
@@ -59,9 +59,6 @@ import           Cardano.Config.Types (DelegationCertFile (..),
                                        Protocol (..), SigningKeyFile (..),
                                        Update (..))
 
--- TODO: consider not throwing this, or wrap it in a local error type here
--- that has proper error messages.
-instance Exception Genesis.ConfigurationError
```
Comment (erikd) [inline cardano-config/src/Cardano/Config/Protocol.hs:64]: :tada: I hate exceptions. Love to see them removed.

File: cardano-node/src/Cardano/CLI/Ops.hs [hunk 2]
```diff
@@ -350,19 +351,19 @@ withRealPBFT
...
-                          Nothing
-                          Nothing
-                          (Just genFile)
-                          nMagic
-                          sigThresh
-                          delCertFp
-                          sKeyFp
-                          update
-  case p of
+  SomeConsensusProtocol cp <-
```
Comment (erikd) [inline cardano-node/src/Cardano/CLI/Ops.hs]: Can we avoid making semantically null whitespace changes? Changes that include whitespace and non-obvious changes in the same PR are more difficult to review.

Comment (dcoutts) [reply cardano-node/src/Cardano/CLI/Ops.hs]: done

**Comment by dcoutts** [comment]
> I can test the scripts once you have rebased.

@Jimbo4350 I added compat aliases for the protocol names in the places where the protocol name was really being interpreted as a file format indicator.

So where scripts used the `cardno-cli` with flags like `--real-pbft`, there is now:
* `--byron-legacy-formats`
* `--byron-formats`
* `--shelley-formats`

The `--real-pbft` is still supported but is now a flag alias hidden from the help output.

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**